x-pack/filebeat/input/gcs/scheduler.go

```diff
@@ -216,6 +223,8 @@ func (s *scheduler) addFailedJobs(ctx context.Context, jobs []*job) []*job {
 		jobMap[j.Name()] = true
 	}
 
+	s.log.Debugf("scheduler: %d failed jobs found", len(s.state.checkpoint().FailedJobs))
```

Comment:
This is racey (actually the `state.checkpoint` method is a race footgun).

Reply:
@efd6, should we lock this also then ?

Reply:
and I checked that in every place where s.state.checkpoint() is called are before any jobs are spawned

Reply:
It's worth understanding whether this can be concurrently updated here.